resources/assets/reactjs/components/List.js

```diff
@@ -0,0 +1,146 @@
+import React, {Component} from 'react';
+import { getList, addItem, updateItem, deleteItem } from './ListFunction';
+
+class List extends Component {
+    constructor(){
+        super()
+        this.state = {
+            id : '',
+            title: '',
+            editDisabled:false,
+            items:[]
+        }
+        this.onSubmit = this.onSubmit.bind(this)
+        this.onChange = this.onChange.bind(this)
+    }
+
+    componentDidMount(){
+        this.getAll();
+    }
+
+    onChange(e){
+        this.setState({
+            [e.target.name]: e.target.value
+        });
+    }
+
+    getAll() {
+        getList().then(data => {
+            this.setState({
+                title: '',
+                items: [...data]
+            },
+            // () => { console.log(this.state.items)
+            // }
+            )
+        })
+    }
+
+    onSubmit(event){
+        event.preventDefault()
+        addItem(this.state.title).then(() => {
+            this.getAll()
+        })
+        this.setState({
+            title: ''
+        })
+    }
+
+    onUpdate(e){
+        e.preventDefault()
+        updateItem(this.state.title, this.state.id).then(() => {
+            this.getAll()
+        })
+        this.setState({
+            title: '',
+            editDisabled:''
+        })
+        this.getAll()
+    }
+
+    onEdit(itemid, e){
+        e.preventDefault()
+
+        var data = [...this.state.items]
+        data.forEach((item,index) => {
+            if(item.id == itemid) {
+                this.setState({
+                    id:item.id,
+                    title:item.title,
+                    editDisabled:true
+                })
+            }
+        })
+    }
+
+    onDelete (val, e){
+        e.preventDefault()
+        deleteItem(val)
+        this.getAll()
+    }
+
+    render() {
+        return (
+            <div className="col-md-12">
+                <form onSubmit={this.onSubmit}>
+                    <div className="form-group">
+                        <label htmlFor="title">Title</label>
+                        <div className="row">
+                            <div className="col-md-12">
+                                <input type="text" className="form-control" id ="title"
+                                name="title" value={this.state.title||''}
```

Comment:
Trước và sau các operator như `+ - * / || &&` thì cũng có 1 dấu cách `value={this.state.title || ''}`